Automattic/wp-calypso
# Focused Launch: Render free sub-domain last

#### Changes proposed in this Pull Request

* When using the `<DomainPicker />` component in the Summary view of the Focused Launch modal we need to have the option of rendering the free wordpress.com sub domain last in the list of suggestions.

#### Testing instructions

Please read the acceptance criteria(s) associated with https://github.com/Automattic/wp-calypso/issues/47748 to ensure they are satisfied by this PR

Also some refactoring to the persistent domain logic has been done as part of this PR. For testing instructions on this please refer to https://github.com/Automattic/wp-calypso/pull/47633

**Focused launch**

1. Go to `/page/SITE_ID_CREATED_WITH_START.wordpress.com/home?flags=create/focused-launch-flow-calypso`
2. Check that the free .wordpress.com sub domain is last in the list of suggestions.
3. Click "View all domains" and check that the free .wordpress.com sub domain remains first in the list of suggestions.
4. Select a premium domain, and go back to the Summary view to check that the free .wordpress.com sub domain is still last in the list of suggestions.

**Gutenboarding - Regression testing**

1. Go to `/new`.
2. Enter a site name. Go to next step which should be the domain step.
3. Your free .wordpress.com sub domain should be first in the list of domains suggestions as seen in the below screenshot

<img width="1487" alt="Screenshot 2020-12-07 at 15 13 51" src="https://user-images.githubusercontent.com/8639742/101368258-d6859900-389e-11eb-8510-712faa7ae88e.png">

**Editing Toolkit Step-by-Step Launch - Regression testing**

1. Checkout this branch and run `npm run dev --sync` from `wp-calypso/apps/editing-toolkit` ensuring you're also logged into your sandbox environment.
2. Go to `/new` and go through the process to create a new site, you will land in the editing experience.
3. Sandbox your newly created site in your host file
4. Flush your cache & DNS and then refresh your site made in step 1 and ensure that you're using the `dev` version of the Editing Toolkit
5. Click "Launch". This should open the launch modal. Proceed to the step "Select a domain" and your free .wordpress.com sub domain should be separated by a heading and first in the list as viewed in the below screenshot

<img width="1381" alt="Screenshot 2020-12-07 at 15 11 54" src="https://user-images.githubusercontent.com/8639742/101368092-a6d69100-389e-11eb-8b31-ca669199113b.png">

Fixes https://github.com/Automattic/wp-calypso/issues/47748
